src/core/server/saved_objects/import/utils.ts

```
@@ -0,0 +1,107 @@
+
+export const getDataSourceTitleFromId = async (
+  dataSourceId: string,
+  savedObjectsClient: SavedObjectsClientContract
+) => {
+  return await savedObjectsClient.get('data-source', dataSourceId).then((response) => {
+    const attributes: any = response?.attributes || {};
+    return attributes ? attributes.title : undefined;
+  });
+};
+
+export const extractVegaSpecFromSavedObject = (savedObject: SavedObject) => {
+  if (isVegaVisualization(savedObject)) {
+    // @ts-expect-error
+    const visStateObject = JSON.parse(savedObject.attributes?.visState);
+    return visStateObject.params.spec;
+  }
+
+  return false;
+};
+
+const isVegaVisualization = (savedObject: SavedObject) => {
+  // @ts-expect-error
+  const visState = savedObject.attributes?.visState;
+  if (!!visState) {
+    const visStateObject = JSON.parse(visState);
+    return !!visStateObject.type && visStateObject.type === 'vega';
+  }
+  return false;
+};
+
+const updateDataObject = (dataObject: any, props: UpdateDataSourceNameInVegaSpecProps) => {
+  const { newDataSourceName, previousDataSourceName } = props;
+  if (dataObject.hasOwnProperty('url') && dataObject.url.hasOwnProperty('index')) {
+    // Update if the object contains the old data source name
+    if (
+      dataObject.url.hasOwnProperty('data_source_name') &&
+      previousDataSourceName &&
+      dataObject.url.data_source_name === previousDataSourceName
+    ) {
+      dataObject.url.data_source_name = newDataSourceName;
+    }
+    // Update if the object fetched data from the local cluster
+    else if (!dataObject.url.hasOwnProperty('data_source_name') && !previousDataSourceName) {
+      dataObject.url.data_source_name = newDataSourceName;
+    }
+  }
+
+  return dataObject;
+};
+
+const parseJSONSpec = (spec: string) => {
+  try {
+    const jsonSpec = JSON.parse(spec);
+
+    if (jsonSpec && typeof jsonSpec === 'object') {
+      return jsonSpec;
+    }
+  } catch (e) {
+    return false;
+  }
+
+  return false;
+};
```

Instead of returning a json when condition meets and return a boolean when failed.
What if directly return undefined when condition not meet?

```
const parseJSONSpec = (spec: string) => {
  try {
    const jsonSpec = JSON.parse(spec);

    if (jsonSpec && typeof jsonSpec === 'object') {
      return jsonSpec;
    }
  } catch (e) {
    return;
  }

  return;
};
```